examples/platforms/utils/power_calibration.cpp

```diff
@@ -0,0 +1,208 @@
+/*
+ *  Copyright (c) 2022, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "power_calibration.h"
```

Comment:
Follow up from other comment about unit test, I have an idea/suggestion:

How about instead of adding this module here in `examples/platforms/utils` we implement this in OT core itself and add it under `src/core/utils`?

We can treat "power calibration" feature similar to how we handle "otPlatSettings" (`Settings`) and "flash":
- We define a bunch of `otPlatSettings{}` platform APIs.
- Platform can choose to implement `otPlatSettings{}` APIs directly (however they want) 
- Or they can enable `OPENTHREAD_CONFIG_PLATFORM_FLASH_API_ENABLE` and then no need to provide `otPlatSettings` APIs and our flash implementation in `src/core/utils` will be used.

Considering similar model for "power calibration":
- We define `otPlatRadio{}` APIs for power calibration in `plratform/radio.h`
  - Platform can choose to implement them directly
  - Or they can choose to use the implementation we provide in OT core.
- We add `OPENTHREAD_CONFIG_RADIO_POWER_CALIBRATION_TABLE_ENABLE` feature in OT: 
   - When enabled, we add the implementation of `PowerCalibration`.
   - We can add also a bunch of configs to set the size of table, size of settings data, etc. (for our OT feature).

Some (small) added benefits of moving `PowerCalibration` in the core:
- We can use all the library and helper classes/methods in OT, `Array`, `Clearable`, `VerifyOrExit`, etc  in the code
- Adding unit test for it fits better (as it is a module in OT).

Thoughts?

Reply:
I think that is a good idea, I have moved the `power_calibration.cpp` to `src/core/utils`.